### PR TITLE
feat: add group preset all patches together

### PR DESCRIPTION
**Pull request by StanGirard**
## Changes

Added the new group preset: `allPatches` in order to group patches together.

## Context

We were creating a default configuration for all of our projects at my company. We wanted to simplify as much as possible the default configuration. We wanted to group patches together in order to reduces CI time, however the preset did not exist. 
Therefor the creation of this new preset

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [X] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
not sure if this is a benefit for many users.

it also only works when you separate minor and patch updates

**Comment by StanGirard** [comment]
Separating minor and patches allows us as SRE to automate security patches of different applications. Might not be useful for developers, but as an SRE, it is a feature that I've used on all my projects.

**Comment by viceice** [comment]
most users don't use that to reduce noise

**Comment by StanGirard** [comment]
I understand the noise issue. Therefore the grouping. We use this present in order to not receive any notification for patches:

```json
"extends": [
    "config:base",
    ":automergePatch"
  ],
  "platformAutomerge": true,
  "automergeType": "branch",
  "packageRules": [
    {
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "patch"
      ],
      "groupName": "all patch dependencies",
      "groupSlug": "all-patch"
    }
  ]
  ```

**Comment by viceice** [comment]
this won't work when there is a minor and a patch update. then you only get the minor PR.

File: lib/config/presets/internal/group.ts [hunk 1]
```diff
@@ -31,6 +31,17 @@ const staticGroups = {
       },
     ],
   },
+  allPatches: {
+    description: 'Group all `patch` updates together.',
+    packageRules: [
+      {
+        matchPackagePatterns: ['*'],
```
Comment (rarkins) [inline lib/config/presets/internal/group.ts:38]:
```suggestion
```
Not needed

File: lib/config/presets/internal/group.ts [hunk 2]
```diff
@@ -31,6 +31,17 @@ const staticGroups = {
       },
     ],
   },
+  allPatches: {
+    description: 'Group all `patch` updates together.',
```
Comment (rarkins) [inline lib/config/presets/internal/group.ts:35]:
```suggestion
    description: 'Group all `patch` updates together. Combine with separateMinorPatch=true',
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Pull request closed without merging**